src/backend/api/routes/api/get_commits_per_day.route.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/python3
+# -*- coding=utf-8 -*-
+r"""
+
+"""
+import fastapi
+import pydantic
+from api.database.models import Repository, Commit
+from api.database.db import createLocalSession
+from sqlalchemy import select, func
+from sqlalchemy.sql import extract
+
+router = fastapi.APIRouter()
+
+class ResponseModel(pydantic.BaseModel):
+    commits_per_day: dict[str, int]
+
+    class Config:
+        orm_mode = True
+
+@router.get("/commits_per_day/{workspace_id}/{year}", response_model=ResponseModel)
```

Comment:
```suggestion
@router.get("/api/private/commits-per-day/{workspace_id}/{year}", response_model=ResponseModel)
```
Bei den sogenannten "slugs" (hier workspace_id und year) besteht immer eine gefahr von naming collisions.
Wenn man beispielsweise eine neue Route erstellen will "/api/private/commits-per-day/with-more-details/", dann würde die Route mit der mit den slugs kolledieren, und wird wahrscheinlich von Router falsch verarbeitet.

Das ist in diesem Fall nicht realistisch, aber trotzdem fände ich query parameters hier einfach praktischer. Die parameter werden mit den slugs ja auch nur "am ende rangehängt", da bieten sie meiner meinung nach in diesem Fall keinen mehrwert gegenüber slugs.

TL;DR: könntest du hier liebr Query Parameters benutzen? Mindestens für das year?

Reply:
done